client/src/Components/FilterSection/index.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useContext, useState } from 'react';
+// import { useLocation } from 'react-router-dom';
+import { Input, Select } from 'antd';
+import { useTranslation } from 'react-i18next';
+import './style.css';
+import { ServiceLocation } from '../../Context/ServiceLocationContext';
+import { locationObject, serviceObject, request } from '../../utils';
+
+function FilterSection() {
+  const { t } = useTranslation();
+  // const locationData = useLocation();
+  // const search : any = locationData.state;
+  // const { serviceSearch, locationSearch, craftsmanSearch } = search;
+
+  const [selectMainService, setMainService] = useState(true);
+  const [subServices, setSubServices] = useState<serviceObject[]>([]);
+  const { data: { location, services } } = useContext(ServiceLocation);
+  const { Option } = Select;
+
+  const handelSelectMainService = async (service: number) => {
+    setMainService(!selectMainService);
+    const { data } : any = await request('get', `/subservices/${service}`);
+    setSubServices(data);
+  };
+
+  return (
+    <div className="filter-container">
+      <h2 className="filter-main-text">{t('filter-heading')}</h2>
+      <span className="filter-input-text">{t('home-search-name')}</span>
+      <Input placeholder={t('search-name')} className="filter-inputs" />
+
+      <span className="filter-input-text">{t('search-by-main-service')}</span>
+      <Select
+        placeholder={t('service')}
+        className="filter-inputs"
+        onSelect={(service: number) => handelSelectMainService(service)}
```

Comment:
input handler shouldn't send a request,
The best thing to do is to set input handler setState for the main service then in useEffect, you can send a request to get sub services based on the main service user select